GaelAlburo/backend-project
# feat(categories): add Dockerfile for containerization

Description:
This PR adds a Dockerfile to containerize the Categories API using a lightweight Alpine-based Python image. The configuration includes security best practices, health checks, and production-ready settings with Gunicorn as the WSGI server.

Changes:
- Add Dockerfile with Python 3.13.0 Alpine base image
- Configure non-root user 'app' for security
- Set up Gunicorn with 4 worker processes
- Add health check endpoint configuration
- Configure port 8000 for production use
- Add curl for container health monitoring

Impact:
- Enables secure containerized deployment using Alpine Linux
- Implements production-grade WSGI server configuration
- Follows security best practices with non-root user
- Provides container health monitoring
- Optimizes image size with Alpine base

Testing:
1. Build Docker image:
   docker build -t categories-api .

2. Run container:
   docker run -p 8000:8000 categories-api

3. Verify container health:
   curl http://localhost:8000/healthcheck

4. Test API endpoints through port 8000:
   - GET /api/v1/categories
   - POST /api/v1/categories
   - PUT /api/v1/categories/<id>
   - DELETE /api/v1/categories/<id>"
